src/main/haskell/kore/src/Kore/AST/Common.hs

```diff
@@ -1085,161 +1085,17 @@ data Pattern level variable child where
-            (ApplicationPattern _, _) -> LT
-            (_, ApplicationPattern _) -> GT
-            (BottomPattern a', BottomPattern b') -> liftCompare liftedCompare a' b'
-            (BottomPattern _, _) -> LT
-            (_, BottomPattern _) -> GT
-            (CeilPattern a', CeilPattern b') -> liftCompare liftedCompare a' b'
-            (CeilPattern _, _) -> LT
-            (_, CeilPattern _) -> GT
-            (DomainValuePattern a', DomainValuePattern b') -> compare a' b'
-            (DomainValuePattern _, _) -> LT
-            (_, DomainValuePattern _) -> GT
-            (EqualsPattern a', EqualsPattern b') -> liftCompare liftedCompare a' b'
-            (EqualsPattern _, _) -> LT
-            (_, EqualsPattern _) -> GT
-            (ExistsPattern a', ExistsPattern b') -> liftCompare liftedCompare a' b'
-            (ExistsPattern _, _) -> LT
-            (_, ExistsPattern _) -> GT
-            (FloorPattern a', FloorPattern b') -> liftCompare liftedCompare a' b'
-            (FloorPattern _, _) -> LT
-            (_, FloorPattern _) -> GT
-            (ForallPattern a', ForallPattern b') -> liftCompare liftedCompare a' b'
-            (ForallPattern _, _) -> LT
-            (_, ForallPattern _) -> GT
-            (IffPattern a', IffPattern b') -> liftCompare liftedCompare a' b'
-            (IffPattern _, _) -> LT
-            (_, IffPattern _) -> GT
-            (ImpliesPattern a', ImpliesPattern b') -> liftCompare liftedCompare a' b'
-            (ImpliesPattern _, _) -> LT
-            (_, ImpliesPattern _) -> GT
-            (InPattern a', InPattern b') -> liftCompare liftedCompare a' b'
-            (InPattern _, _) -> LT
-            (_, InPattern _) -> GT
-            (NextPattern a', NextPattern b') -> liftCompare liftedCompare a' b'
-            (NextPattern _, _) -> LT
-            (_, NextPattern _) -> GT
-            (NotPattern a', NotPattern b') -> liftCompare liftedCompare a' b'
-            (NotPattern _, _) -> LT
-            (_, NotPattern _) -> GT
-            (OrPattern a', OrPattern b') -> liftCompare liftedCompare a' b'
-            (OrPattern _, _) -> LT
-            (_, OrPattern _) -> GT
-            (RewritesPattern a', RewritesPattern b') -> liftCompare liftedCompare a' b'
-            (RewritesPattern _, _) -> LT
-            (_, RewritesPattern _) -> GT
-            (StringLiteralPattern a', StringLiteralPattern b') -> compare a' b'
-            (StringLiteralPattern _, _) -> LT
-            (_, StringLiteralPattern _) -> GT
-            (CharLiteralPattern a', CharLiteralPattern b') -> compare a' b'
-            (CharLiteralPattern _, _) -> LT
-            (_, CharLiteralPattern _) -> GT
-            (TopPattern a', TopPattern b') -> liftCompare liftedCompare a' b'
-            (TopPattern _, _) -> LT
-            (_, TopPattern _) -> GT
-            (VariablePattern a', VariablePattern b') -> compare a' b'
+    liftCompare liftedCompare a b = $(makeLiftCompare ''Pattern) liftedCompare a b
```

Comment:
Just curious: You removed (say) `pat` from the argument list of `liftShowsPrec` but you didn't remove `a` and `b` here. Is there a reason for this difference?